### PR TITLE
Tidy up some unused components

### PR DESCRIPTION
**Pull request by joegallo**
Replaces #93557

A bit of followup from #93522 and #93438, I noticed some unused components hanging around, so I did a round of tree-shaking to see what could be dropped.

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1473,20 +1383,12 @@
          <artifact name="tini-0.19.0-amd64.jar">
             <sha256 value="93dcc18adc78c65a028a84799ecf8ad40c936fdfc5f2a57b1acda5a8117fa82c" origin="Generated by Gradle"/>
          </artifact>
-         <artifact name="tini-0.19.0-arm64.jar">
```
Comment (joegallo) [inline gradle/verification-metadata.xml:1476]: It's very interesting to me that we had two different `sha256` values here.

Comment (mark-vieira) [reply gradle/verification-metadata.xml:1476]: This is intentional. One is the x86 artifact and the other is for arm. These are used on our docker images and both are needed.

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -4122,10 +3589,5 @@
             <sha256 value="282ae185fc2ff27da7714af9962897c09cfefafb88072219c4a2f9c73616c026" origin="Generated by Gradle"/>
          </artifact>
       </component>
-      <component group="zulu_8" name="macos" version="8u302">
```
Comment (mark-vieira) [inline gradle/verification-metadata.xml:4125]: This is also needed for some testing using very old versions of ES that require Java 8.

File: gradle/verification-metadata.xml [hunk 3]
```diff
@@ -3229,11 +2711,6 @@
             <sha256 value="e469bd39f936999f256002631003ff022a22951da9d5bd9789c7abfa9763a292" origin="Generated by Gradle"/>
          </artifact>
       </component>
-      <component group="org.bouncycastle" name="bctls-fips" version="1.0.9">
```
Comment (mark-vieira) [inline gradle/verification-metadata.xml:3232]: This jar is needed, but only when `tests.fips.enabled` system property is set to `true`.

**Review by mark-vieira — COMMENTED** [review]
How did you identify the items to remove? The tricky thing with this file is if you regenerate it from scratch, it'll only include artifacts that were resolved in the current build. We have lots of different possible tasks to execute that all require different artifacts and some are even host platform specific so it's hard to create a single build that constitutes all possible required artifacts. That's why creating the iniital file was a bit of a game of trial and error.

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (joegallo) [reply gradle/verification-metadata.xml:1476]: OMG, the number of times I read that as `amd64` *twice*. Yarp, they're different. 😵‍💫

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (joegallo) [reply gradle/verification-metadata.xml:4125]: Do we still test on `macos`?

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (joegallo) [reply gradle/verification-metadata.xml]: Added back via 09c5e430fc0e769607509b6ef5a441bfe3ee0a0c

File: gradle/verification-metadata.xml (hunk 3 above)
Comment (joegallo) [reply gradle/verification-metadata.xml]: Added back via 6010970617be517ec5dd32ab3bead392cbb0d2be

**Comment by joegallo** [comment]
@elasticmachine run elasticsearch-ci/rest-compatibility

**Comment by joegallo** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by joegallo** [comment]
@elasticmachine run elasticsearch-ci/part-3

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (joegallo) [reply gradle/verification-metadata.xml]: Added back via c51a4326e6ea06b9b1e68d0af9f7d865ffc93253

**Review by mark-vieira — APPROVED** [review]

**Pull request merged**